### PR TITLE
Pass RT mode to LGC and call SetVsp

### PR DESCRIPTION
**Pull request by LLJJDD**
1. Tell LGC about the RT indirect mode, so that it can determine which passes
to use.

2. When continuations is enabled, use lgc.cps.set.vsp to setup stack pointer
in SpirvLowerRayTracing pass.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 38fb2aa8c608fbc14885cf67203f7e251cc3ddbd
<details><summary>CTS tests (Failed: 0/209067)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34735/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35053/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38620/69504 (55.6%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 30884/69504 (44.4%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/interface/lgc/Pipeline.h [hunk 1]
```diff
@@ -103,6 +103,14 @@ enum class ThreadGroupSwizzleMode : unsigned {
   Count,
 };
 
+// Enumerate the ray tracing indirect modes.
+enum class RayTracingIndirectMode : unsigned {
+  NotIndirect = 0,    // Not in indirect mode (or not ray tracing pipeline)
+  Legacy = 1,         // Legacy indirect mode
+  Continuations0 = 2, // Continuations flow that based on Continufy pass
```
Comment (Flakebi) [inline lgc/interface/lgc/Pipeline.h:110]: Instead of 0 and 1, maybe name this one `ContinuationsContinufy`?

Comment (LLJJDD) [reply lgc/interface/lgc/Pipeline.h:110]: Makes sense, what about the other one, just `Continuations`?

Comment (Flakebi) [reply lgc/interface/lgc/Pipeline.h:110]: Yeah, I think just `Continuations` is fine.

**Comment by amdvlk-admin** [comment]
### Test summary for commit c101aaa069e782cf405be9eb2d4905c72da54b7a
<details><summary>CTS tests (Failed: 0/209351)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34735/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35053/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38620/69788 (55.3%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 31168/69788 (44.7%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Review by Flakebi — APPROVED** [review]

**Review by nhaehnle — APPROVED** [review]
LGTM

**Pull request merged**